### ISSUE TITLE
Question abou server management

### ISSUE THREAD

**Issue by Methes** [GTNewHorizons/NewHorizons#4131]
I'm an amateur regarding server hosting so would like to ask you guys if you've considered running multilple servers for the same map with player transition between the servers. Would this increase performance? Something like Bungeecord?
The idea is you have the same map (same seed) divided by multiples of region files into different servers with same playerdata. Meaning a travelling player would eventually reach a border of certain server and would (ideally seamlessly) transition to next server with same character.
In my understanding the MC server behaves better when run multiple times for fewer players especially with multicore CPUs.
Thank you for your insights!

**Comment by Dream-Master** [comment]
@Namikon know it maybe better but so far i know is Bungeecord with so many mods we have not possible.

**Comment by Namikon** [comment]
Yes, we've considered this. And we actually had several test-setups for this purpose.
And yes, It definitely increases the overall server performance. The main Minecraft server is "loop based" and processes everything in order, one task after another. As such, you cannot split the load between multiple cores. And the more dimensions you have, the more things have to be calculated within the 50ms TimeSlot that is required to keep up the 20 Ticks per Second.
One solution to spread this load is to use BungeeCord to combine multiple Servers to one large one.

What you are describing is what several MC communities did with Packs on 1.6.4 and earlier; And what most vanilla Servers probably are still doing; One Server for Building your home, one for resource mining, probably one Server for ~5 Void dimensions, etc.

The Problem why we don't use this setup is that we have a lot of mods which store player data.
In order to Setup a BungeeCord Server-Cluster, you'd have to find/develop a proper data-sync function for every mod that is using/storing Playerdata in *any* way; Without opening up item duplication loopholes. (Like; Move Inventory/Backpack/... from Server A to B, verify that it has arrived, then remove the source on Server B to prevent duplication. And all that within the few seconds required to switch the Servers)

Another issue is stuff like EnderChests/Tanks, any "wireless", cross dimensional item/data transport. (Wireless AE with infinity card,..) etc. You'd have to either just disable them, or provide a way to properly transfer items/data from server A to B (Again without duplicating). Might work in some way for items, but with AE?

Thaumcraft is also often mentioned as a reason why BungeeCord isn't possible. It has been fixed, so you actually can transition between Servers now, but it still has issues like a broken Thaumometer until you relog on the target Server. 

I haven't tried the Setup with a full modpack yet; I've stopped after running several attempts with a very simple 1.7.10 setup; Gregtech, Thaumcraft, Tinkers, NEI and a few basic mods. Already didn't work properly.

TL;DR:
Yes, it is possible. Given enough time and people to analyze the mods and develop a proper data-sync, either by code changes, or file copy across servers, remote-mounted filesystems,...

Also later on, after the mods have been changed, you need enough skilled people to monitor/fix the individual Servers in that cluster and keep them in sync. 

If we could, we'd love to set up such a cluster. But at the moment, we don't have the resources for such a project.

**Comment by Methes** [comment]
Thank you very much for finding time to write this detailed and well explained answer. Thanks!
